### DISCUSSION TITLE
[httpfs] add or document how to log http query launched by duckdb client

### DISCUSSION THREAD

**Discussion by alexisdondon** [duckdb/duckdb#10547] [Ideas]
### What happens?

We use duckdb on a minio instance and the next commands is very long to execute some several minutes/hours to see query incoming on minio.

Writing/reading to parquet works like a  charm, it seems to be related to PARTITION BY statement and http query launched in the process.

To dig into and check where could be the problem( maybe relative to our network configuration), this is quite difficult without flag to allow more information on what's duckdb is doing/waiting for.

Browsing the documentation i do not have seen some flags verbose debug mode to allow logging what's going on.

```
INSTALL httpfs;
LOAD httpfs;
CREATE TABLE mytable as select * from (values ('a','1'),('b','2')) as t(col1,col2);
SET s3_url_style='path';
COPY mytable TO 's3://bucket/table' (FORMAT PARQUET, PARTITION_BY (col1));
```
is there some way to log current http queries and retries duckdb is currently doing?

(our minio implementation do not support vhost style, trying on a minio instance that support vhost style it is working however i am not sur that this is the root cause that's why logging the http query on duckdb client could allow me a better understanding on what's going on)

### To Reproduce

```
INSTALL httpfs;
LOAD httpfs;
CREATE TABLE mytable as select * from (values ('a','1'),('b','2')) as t(col1,col2);
SET s3_url_style='path';
COPY mytable TO 's3://bucket/table' (FORMAT PARQUET, PARTITION_BY (col1));
```

### OS:

windows and ubuntu

### DuckDB Version:

v0.9.2

### DuckDB Client:

python and client

### Full Name:

Alexis D

### Affiliation:

Insee

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mause** [comment]
This is more a feature request than a bug, so I'll move it there

**Reply by alexisdondon** [reply]
Yes but i don't know it there is really no bug on this about vhost style and path style in this case.

**Reply by alexisdondon** [reply]
Digging little it seems related to https://github.com/duckdb/duckdb/issues/7167 .

**Reply by alexisdondon** [reply]
![image](https://github.com/duckdb/duckdb/assets/63847345/9b28ccff-90aa-4888-a398-e6edf37b2a6f)

**Reply by alexisdondon** [reply]
Reallly strange this call to bucketname.s3.amazonaws.com. i have set s3_url_style and s3_endpoint on another domain, and it seems that there is some 443 https request to some default s3 domain.

If true and reproductible :
 - should be handle with care as for a data framework launching query like that is very sensitive.
 - could explain my latency problem as this domain can't be reachable depend on network configuration
 
 Allow logging this query on duckdb client side would make the debugging easier and probably allow more transparency.

**Comment by alexisdondon** [comment]
@Mause I think underlying behaviour is a bug. something have been done related to this https://github.com/duckdb/duckdb/issues/7167 ?
if yes, something equivalent need to be done when PARTITION BY statement (probably multipart upload) is in the process?

**Reply by alexisdondon** [reply]
Launching this script in an offline environment we can see that there is outside request to amazon send each 30s with retry 4 during looooong minutes until don't know why duckdb take care of s3 configuration
